R/factorialAnalysis.R

```diff
@@ -15,115 +15,126 @@
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #
 
-factorialAnalysis <- function(jaspResults, dataset, options, ...) {
+factorialAnalysis <- function(jaspResults, dataset, options, ...){
 
-  ready <- (length(options[["FAassignedFactors"]]) >= 2 && !is.null(options[["FAresponse"]]))
+  ready <- (length(options[["FAassignedFactors"]]) >= 2 && !is.null(options[["FAresponse"]]) && !is.null(options[["FArunOrder"]]))
 
-  if (ready)
+  if(ready)
     dataset <- .factorialAnalysisReadData(dataset, options)
 
-  if (is.null(jaspResults[["factorialRegressionANOVA"]]))
-    .factorialRegressionANOVAcreateTable(jaspResults, options)
+  fit <- .factorialRegression(jaspResults, dataset, options, ready)
 
-  if (is.null(jaspResults[["factorialRegressionCoefficients"]]))
-    .factorialRegressionCoefficientsCreateTable(jaspResults, options)
-
-  .factorialRegression(jaspResults, dataset, options, ready)
+  if(is.null(jaspResults[["factorialRegressionANOVA"]]))
+    .factorialRegressionANOVAcreateTable(jaspResults, options, ready, fit)
 
+  if(is.null(jaspResults[["factorialRegressionCoefficients"]]))
+    .factorialRegressionCoefficientsCreateTable(jaspResults, options, ready, fit)
 }
 
-.factorialAnalysisReadData <- function(dataset, options) {
+.factorialAnalysisReadData <- function(dataset, options){
 
-  if (!is.null(dataset))
+  if(!is.null(dataset))
     return(dataset)
   else
-    return(.readDataSetToEnd(columns.as.numeric = c(options[["FArunOrder"]],
-                                                    options[["FAresponse"]]),
+    return(.readDataSetToEnd(columns.as.numeric = c(options[["FArunOrder"]], options[["FAresponse"]]),
                              columns.as.factor = options[["FAassignedFactors"]]))
 
 }
 
-.factorialRegression <- function(jaspResults, dataset, options, ready, ...) {
+.factorialRegression <- function(jaspResults, dataset, options, ready, ...){
 
-  if (!ready)
+  if(!ready)
     return()
 
-  factors <- unlist(dataset[,options[["FAassignedFactors"]]])
-  factors <- data.frame(factors)
-  response <- unlist(dataset[,options[["FAresponse"]]])
+  factors <- unlist(dataset[,options[["FAassignedFactors"]]], use.names = FALSE)
+  response <- unlist(dataset[,options[["FAresponse"]]], use.names = FALSE)
+
+  print(1:100)
+  names <- print(colnames(decodeColNames(dataset[options[["FAassignedFactors"]]])))
```

Comment:
1. Why `print`?
2. Why `decodeColNames`? Can your code handle all possible characters in all possible encodings? So also things like "_𝝘_ꙃ__ቈ__ᳯ__㉹"? If not, or if you didn't test that, then you probably don't want to use this here.

Reply:
✅